app/src/main/java/com/team300/fridge/FoodItemDialog.java

```diff
@@ -0,0 +1,92 @@
+package com.team300.fridge;
+
+import android.app.Dialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.support.v7.app.AlertDialog;
+import android.support.v7.app.AppCompatDialogFragment;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+
+import java.time.LocalDate;
+
+public class FoodItemDialog extends AppCompatDialogFragment {
+    private EditText editTextQuantity;
+    private EditText editTextDate;
+    private FoodItemDialogListener mListener;
+
+    @Override
+    public Dialog onCreateDialog(Bundle savedInstanceState) {
+        String name = getArguments().getString("Food_Item_Name");
+        //TODO: Once we have brand information show that as well
+        //getArguments().getString("Food_Item_Brand");
+        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
+        View view = getActivity().getLayoutInflater().inflate(R.layout.layout_food_item_dialog, null);
+        builder.setView(view)
+                .setTitle(name)
+                .setNegativeButton("cancel", (dialog, which) -> {})
+                .setPositiveButton("confirm", (dialog, which) -> {});
+
+        editTextQuantity = view.findViewById(R.id.edit_quantity);
+        editTextDate = view.findViewById(R.id.edit_date);
+
+        AlertDialog dialog = builder.create();
+        dialog.show();
+        Button button = dialog.getButton(DialogInterface.BUTTON_POSITIVE);
+        button.setOnClickListener(new CustomListener(dialog, name));
+        dialog.dismiss();
+        return dialog;
+    }
+
+    //We need a custom OnClickListener to be able to validate input before dismissing the dialog
+    public class CustomListener implements View.OnClickListener {
+        private final Dialog dialog;
+        private String name;
+
+        public CustomListener(Dialog dialog, String name) {
+            this.dialog = dialog;
+            this.name = name;
+        }
+        @Override
+        public void onClick(View v) {
+            String date = editTextDate.getText().toString();
+            if (validate(date)) {
```

Comment:
Make sure that they can't add 0 quantity. I tried to do that and it crashed the app

Reply:
I also tried a bad date, it gave me the error as expected, but then it crashed once I put in valid inputs for the same item